File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/types/digitaltwin/DigitalTwinDescriptor.java
package org.opcfoundation.webserver.types.digitaltwin;

import org.eclipse.milo.opcua.stack.core.types.builtin.LocalizedText;
import org.eclipse.milo.opcua.stack.core.types.builtin.NodeId;
import org.opcfoundation.webserver.digitaltwin.digitaltwin.DigitalTwinType;

public class DigitalTwinDescriptor {
    private final String id;
    private final LocalizedText displayName;
    private final NodeId typeId;

    public DigitalTwinDescriptor(
            String id,
            LocalizedText displayName,
            DigitalTwinType digitalTwinType)
    {
        this.id = id;
        this.displayName = displayName;
        this.typeId = digitalTwinType.nodeId();
    }

    public String getId() {
        return id;
    }

    public LocalizedText getDisplayName() {
        return displayName;
    }

    public NodeId getTypeId() {
        return typeId;
    }
}

File: opcua-webservice-node/source/src/server/types/digitaltwin/DigitalTwinDescriptor.ts
import { UaLocalizedText, UaNodeId } from "opcua-webapi-ts";
import { UaObjectType } from "../../addressspace/nodes/UaObjectType";

export class DigitalTwinDescriptor {
    private readonly _id: string;
    private readonly _displayName: UaLocalizedText;
    private readonly _typeId: UaNodeId;

    constructor(
        id: string, 
        displayName: UaLocalizedText, 
        digitalTwinType: UaObjectType) {
        this._id = id;
        this._displayName = displayName;
        this._typeId = digitalTwinType.nodeId;
    }

    get id(): string {
        return this._id;
    }

    get displayName(): UaLocalizedText {
        return this._displayName;
    }

    get typeId(): UaNodeId {
        return this._typeId;
    }
}
